reconcile/terraform_aws_route53.py

```diff
@@ -116,6 +119,50 @@ def build_desired_state(zones):
                 logging.debug(msg)
                 record["records"] = record_values
 
+            # Process '_target_namespace_zone'
+            target_namespace_zone = record.pop("_target_namespace_zone", None)
+            if target_namespace_zone:
+                tf_resources = target_namespace_zone["namespace"]["terraformResources"]
+                tf_zone_name = target_namespace_zone["name"]
+                tf_zone_resources = [
+                    tfr
+                    for tfr in tf_resources
+                    if tfr["provider"] == "route53-zone" and tfr["name"] == tf_zone_name
+                ]
+                if not tf_zone_resources:
+                    logging.error(
+                        f"{zone_name}: field `_target_namespace_zone` found "
+                        f"for record {record_name}, but target zone not found: "
+                        f"{tf_zone_name}"
+                    )
+                    sys.exit(ExitCodes.ERROR)
```

Comment:
Not sure, but would it make sense to raise an exception here and handle this in the caller?

Reply:
NVM. I see the same style is used in the beginning of the function.